designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentService.scala

```diff
@@ -446,38 +447,83 @@ class DeploymentService(
   )(implicit user: LoggedUser, freshnessPolicy: DataFreshnessPolicy): Future[ProcessState] = {
     dbioRunner.run(for {
       inProgressActionNames <- actionRepository.getInProgressActionNames(processDetails.processId)
-      result                <- getProcessState(processDetails, inProgressActionNames, None)
+      result                <- getProcessState(processDetails, inProgressActionNames, None, None)
     } yield result)
   }
 
   override def enrichDetailsWithProcessState[F[_]: Traverse](processTraverse: F[ScenarioWithDetails])(
       implicit user: LoggedUser,
       freshnessPolicy: DataFreshnessPolicy
   ): Future[F[ScenarioWithDetails]] = {
+    val scenarios = processTraverse.toList
     dbioRunner.run(
       for {
-        actionsInProgress <- getInProgressActionTypesForProcessTraverse(processTraverse)
+        actionsInProgress <- getInProgressActionTypesForScenarios(scenarios)
+        // DeploymentManager's may support fetching state of all scenarios at once (StateQueryForAllScenariosSupported capability)
+        // For those DM's we prefetch the state and then use it for resolving individual scenarios.
+        prefetchedStates <- DBIO.from(getPrefetchedStatesForSupportedManagers(scenarios))
```

Comment:
`F` is sometimes the `Id` (this method is used in context of fetching one scenario status. To fetch status of one scenario you ask all deployment managers in every processing type for a status of every scenario. Is the intention of this change? :)

Reply:
- State was only pre-fetched from DM's resposible for scenarios, that were fetched. So if the query was only about periodic scenarios, then only PeriodicDM was queried about all states
- I made a bold assumption, that fetching all states is not heavier than fetching a single on. It is true for Flink and Periodic, but may not be true in general
- I modified the implementation, to only pre-fetch statuses from DM, if there is more than one scenario handled by that DM in the current query.